File: src/main/java/me/pcasaes/hexoids/configuration/QueueMetricsProvider.java
package me.pcasaes.hexoids.configuration;

import jakarta.enterprise.context.ApplicationScoped;
import jakarta.enterprise.inject.Produces;
import jakarta.inject.Singleton;
import me.pcasaes.hexoids.infrastructure.disruptor.QueueMetric;

import java.util.List;

@ApplicationScoped
public class QueueMetricsProvider {

    @Produces
    @Singleton
    public List<QueueMetric> getMetrics() {
        return QueueMetric.getAllMetrics();
    }

}

File: src/main/kotlin/me/pcasaes/hexoids/configuration/QueueMetricsProvider.kt
package me.pcasaes.hexoids.configuration

import jakarta.enterprise.context.ApplicationScoped
import jakarta.enterprise.inject.Produces
import jakarta.inject.Singleton
import me.pcasaes.hexoids.infrastructure.disruptor.QueueMetric

@ApplicationScoped
class QueueMetricsProvider {
    @Produces
    @Singleton
    fun getMetrics(): List<QueueMetric> {
        return QueueMetric.getAllMetrics()
    }
}
